static/js/controllers/contacts.js

```diff
@@ -98,6 +108,7 @@ var _ = require('underscore'),
               contacts.splice(homeIndex, 1);
               contacts.unshift(usersHomePlace);
             } else if (!$scope.filters.search && !$scope.filters.simprintsIdentities) {
+              additionalListItem = true;
```

Comment:
I have updated the logic to treat the case of receiving the homeplace in a subsequent pagination query. Except, as the code is written at this moment, this should never happen, because the search results are filtered so they are "children" of the homeplace by default. 
(See: 
`type = ContactSchema.getChildPlaceType(usersHomePlace.type);` 
and 
 `defaultTypeFilter = { types: { selected: [ type ] }};`).

The only case where homeplace could indeed be received as a Search result is when the user is doing actual filtering, in which case homeplace is not pushed at the top of the list in the first place.